src/popup/pages/NetworkForm.vue

```diff
@@ -97,120 +106,124 @@ export default defineComponent({
+        placeholder: t('pages.network.networkMiddlewarePlaceholder'),
+        label: t('pages.network.networkMiddlewareLabel'),
         dataCy: 'middleware',
       },
       {
         key: 'compilerUrl',
-        placeholder: root.$t('pages.network.networkCompilerPlaceholder'),
-        label: root.$t('pages.network.networkCompilerLabel'),
+        placeholder: t('pages.network.networkCompilerPlaceholder'),
+        label: t('pages.network.networkCompilerLabel'),
         dataCy: 'compiler',
       },
       {
         key: 'backendUrl',
-        placeholder: root.$t('pages.network.backendUrlPlaceholder'),
-        label: root.$t('pages.network.backendUrlLabel'),
+        placeholder: t('pages.network.backendUrlPlaceholder'),
+        label: t('pages.network.backendUrlLabel'),
       },
     ];
 
     const error = ref({});
 
-    const hasErrors = computed(() => (root as any).$validator.errors.items?.length);
-
     const buttonDisabled = computed(
       () => !Object.keys(NETWORK_PROPS).every(
         (key) => !!newNetwork.value[key as keyof INetworkBase],
-      )
-      || !!hasErrors.value,
+      ),
     );
 
     function goBack() {
-      root.$router.push({ name: ROUTE_NETWORK_SETTINGS });
+      router.push({ name: ROUTE_NETWORK_SETTINGS });
     }
 
     function validatorRules(key: keyof INetworkBase) {
       return key === 'name' ? {
+        required: true,
         network_name: true,
         network_exists: [newNetwork.value?.index, networks.value],
-        max: NETWORK_NAME_MAX_LENGTH,
+        max_len: NETWORK_NAME_MAX_LENGTH,
       } : {
         required: true,
         invalid_hostname: true,
       };
     }
 
     async function addOrUpdateNetwork() {
-      if (await $validator.validateAll()) {
-        root.$store.commit('setUserNetwork', {
-          ...newNetwork.value,
-          index: newNetwork.value.index,
-        });
-        await selectNetwork(newNetwork.value.name);
-        goBack();
-      }
+      store.commit('setUserNetwork', {
+        ...newNetwork.value,
+        index: newNetwork.value.index,
+      });
+      await selectNetwork(newNetwork.value.name);
+      goBack();
     }
 
     onMounted(async () => {
-      const { params, query } = root.$route;
+      const { params, query } = route;
+      fieldsToCheckOnMount.value = Object.keys(query);
 
       if (isEdit) {
-        newNetwork.value = { ...networks.value[params.name] };
+        newNetwork.value = { ...networks.value[params.name.toString()] };
       } else if (Object.keys(query).length) {
         // Fields that values are allowed to be passed from the URL query to the form model
         const keys: (keyof INetworkBase)[] = ['name', 'url', 'middlewareUrl', 'compilerUrl'];
         keys.forEach((key) => {
           const val = query[key];
 
           if (val && typeof val === 'string') {
-            set(newNetwork.value, key, val);
+            // @ts-expect-error key is a valid key of INetworkBase
```

Comment:
If key is a valid key of INetworkBase I think it's better to do `[key as keyof INetworkBase]` instead of silencing the TS errors.